### DISCUSSION TITLE
Fargate - Vary the input file per container

### DISCUSSION THREAD

**Discussion by giuliodali** [artilleryio/artillery#1998] [Artillery]
Hi - We have a test where we have an input file of user data for 20,000 users. We want to simulate them all doing different things. 

We've configured an arrival count of 1,000 sessions, over 20 containers and we'd like each container to work a different set of 1,000 users. However, all containers work the first set of 1,000 users. 

How can we tell Artillery to evenly distribute the input file across the containers? So in our case, 1,000 unique user per container.

**Comment by hassy** [comment]
Hi @giuliodali 👋 We don't have support for distributing input files like that right now. The best way to do it right now would be to use some form of shared state between workers, and load subsets of data at runtime (e.g. in a `beforeScenario` hook). A Redis instance could work well for example.

**Reply by Malvika-sys** [reply]
@hassy @giuliodali   Can anyone help me how do we authenticate from local machine to connect automation framework to AWS FARGATE : provide AWS key id and secrets?

**Reply by bernardobridge** [reply]
Hi @Malvika-sys!

I'm assuming you already have an AWS account set up from your org. If that's the case, then you'll need to find the method of authentication to AWS that your organisation typically uses. In [our documentation](https://www.artillery.io/docs/load-testing-at-scale/aws-fargate), we have a link to this [guide from AWS](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html): It has guides to different ways of configuring the AWS credentials setup, and their order of precedence in how the AWS SDK's pick them up.

---

Since you're mentioning setting an AWS key id and secret, I'm assuming you're using an IAM user directly. Most likely you'll need to have the `aws cli` installed on your machine and then run `aws configure` to configure those credentials. (https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html). Another option is to export them as environment variables before running your Artillery script: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html. This is easier, but only works for that particular shell session, so you'd need to do this every time.

_Note: I'm making some assumptions here on your setup. Please consult the AWS documentation I linked above for more information that may work better for your specific setup._
